pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -784,3 +789,32 @@ func (manager *gceServiceManager) getProjectsAPIEndpointAlpha() string {
 
 	return projectsApiEndpoint
 }
+
+// GetAllCurrentZones returns all the zones in which nodes are currently running
+// TODO: Caching, but this is currently only called when we are creating a volume,
+// which is a relatively infrequent operation, and this is only 1 API call.
+// Ideally we want to make this watch-based
+// TODO: When node labels are secure we should go back to using gcloud API
```

Comment:
IIUC, it's actually the K8 Node labels we want to secure and use in the end, not the gce instance labels.